### ISSUE TITLE
Pigeons.json() array input issue

### ISSUE THREAD

**Issue by UBCYujia** [Julia-Tempering/Pigeons.jl#235]
Pigeons.json() creates correct form of json string when taking scalar or 1d array as input. However, when it comes to 2d or higher dimensional arrays, it creates invalid json string (missing comma or "]").

**Comment by miguelbiron** [comment]
Thanks @UBCYujia ! Can you post a minimal reproducible example please?

**Comment by alexandrebouchard** [comment]
Array of arrays at least seem ok:

```
julia> Pigeons.json(test = [[12, 12], [1, 2]])
"{\"test\" : [[12, 12], [1, 2]]}"
```

**Comment by alexandrebouchard** [comment]
But indeed matrix would probably create parsing error:

```
julia> Pigeons.json(test = [1 2; 3 4])
"{\"test\" : [1 2; 3 4]}"
```

Converte the latter into the former?

Thanks for reporting this Yujia!

**Comment by miguelbiron** [comment]
I wonder if we can just use the JSON.jl pkg.

**Comment by UBCYujia** [comment]
Yes I tried JSON.jl and it works well with Pigeons!

**Comment by miguelbiron** [comment]
Nice! I see that they dont support NamedTuples which is the way we use Pigeons.json. But they do support Dicts, so the one liner would be
```julia
Pigeons.json(; kwargs...) = JSON.json(Dict(pairs(kwargs)))
```
or something similar

**Comment by miguelbiron** [comment]
actually `pairs` is enough
```julia
> myjson(; kwargs...) = JSON.json(pairs(kwargs))
> myjson(;a=1,b=2)
"{\"a\":1,\"b\":2}"
```

**Comment by alexandrebouchard** [comment]
@UBCYujia  would you mind creating a PR to fix this issue?

**Comment by UBCYujia** [comment]
Sure! I can try to do that!

**Comment by alexandrebouchard** [comment]
Thank you, and let us know if you have any questions on how to do that. I think Lucas did one earlier so he might be able to assist with the logistics parts of the procedure

**Comment by alexandrebouchard** [comment]
Can we link the merged PR to this and close this issue?

**Comment by UBCYujia** [comment]
Sure! But since that was a very simple fix so I did not create a new branch or PR for it.   
-- Fixed in commit 43487c1

**Comment by alexandrebouchard** [comment]
Thank you!
